internal/indexer/postgres/setindexfinished.go

```diff
@@ -4,83 +4,74 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/jmoiron/sqlx"
-
 	"github.com/quay/claircore"
 	"github.com/quay/claircore/internal/indexer"
 )
 
-func setScanFinished(ctx context.Context, db *sqlx.DB, sr *claircore.IndexReport, scnrs indexer.VersionedScanners) error {
+func (s *store) SetIndexFinished(ctx context.Context, ir *claircore.IndexReport, scnrs indexer.VersionedScanners) error {
 	const (
-		selectScanner = `
-		SELECT id
-		FROM scanner
-		WHERE name = $1
-		  AND version = $2
-		  AND kind = $3;
-		`
-		insertManifestScanned = `
-		WITH manifests AS (
-			SELECT id AS manifest_id
-			FROM manifest
-			WHERE hash = $1
-		)
-		INSERT
-		INTO scanned_manifest (manifest_id, scanner_id)
-		VALUES ((SELECT manifest_id FROM manifests),
-				$2);
-		`
-		upsertIndexReport = `
-		with manifests as (
-			select id as manifest_id from manifest
-			where hash = $1
+		cte = `
```

Comment:
I'm not a fan of building the query like this. It makes it difficult to copy and paste into a database IDE for troubleshooting. It's also a little confusing to piece together the elements.